Userland/Libraries/LibJS/Runtime/RegExpConstructor.h

```diff
@@ -6,10 +6,54 @@
 
 #pragma once
 
+#include <AK/HashMap.h>
+#include <AK/Utf16View.h>
 #include <LibJS/Runtime/NativeFunction.h>
 
 namespace JS {
 
+static Utf16String empty_string(String::empty().view());
+
+// https://github.com/tc39/proposal-regexp-legacy-features#regexp
+// The %RegExp% instrinsic object, which is the builtin RegExp constructor, has the following additional internal slots:
+// [[RegExpInput]]
+// [[RegExpLastMatch]]
+// [[RegExpLastParen]]
+// [[RegExpLeftContext]]
+// [[RegExpRightContext]]
+// [[RegExpParen1]] ... [[RegExpParen9]]
+class RegExpStaticProperties {
+public:
+    Utf16String& input() { return m_input; }
+    Utf16String& last_match() { return m_last_match; }
+    Utf16String& last_paren() { return m_last_paren; }
+    Utf16String& left_context() { return m_left_context; }
+    Utf16String& right_context() { return m_right_context; }
+    HashMap<size_t, Utf16String>& parens() { return m_parens; }
+    Utf16String& paren_at(size_t index)
```

Comment:
Please prefer const getters, and add both const and non-const variants if we need that